docs/next/modules/en/pages/troubleshooting/troubleshooting.adoc

```diff
@@ -0,0 +1,300 @@
+
+The helm chart exposes values for increasing the log level via the usual `values.yaml` configuration parameters:
+
+* Cluster API Operator - log level for CAPI operator can also be increased using helm chart, if installed with Rancher Turtles chart use:
+
+[source,bash]
+----
+        --set cluster-api-operator.logLevel=5
+        --set rancherTurtles.managerArguments[0]="-v=5"
+----
+
+For example:
+
+[source,bash]
+----
+helm upgrade rancher-turtles turtles/rancher-turtles \
+            -n rancher-turtles-system \
+            --set "rancherTurtles.managerArguments={--insecure-skip-verify,-v=5}" \
+            --set cluster-api-operator.logLevel=5
+----
+
+
+* Cluster API Providers - edit CAPIProvider resources for providers where increasing log level is needed. Change to desidered level:
+
+[source,bash]
+----
+CAPIProvider.Spec.Manager.Verbosity=5
+----
+(5 is equivalent to DEBUG)
+
+== How to collect infortmation from Rancher Turtles and CAPI
+
+https://github.com/crust-gather/crust-gather[crust-gather] is a project created by Cluster API developers specifically designed for gathering logs and resource states from CAPI environments. It's a safe and official tool that can be used in any type of environment (development, testing, or production) to collect comprehensive diagnostic information for troubleshooting.
+
+You can install it via following instructions:
+
+[source,bash]
+----
+kubectl krew install crust-gather
+
+kubectl crust-gather --help
+----
+
+Alternatively, it can be installed standalone via *install.sh* script:
+
+[source,bash]
+----
+curl -sSfL https://github.com/crust-gather/crust-gather/raw/main/install.sh | sh
+
+crust-gather --help
+----
+
+You can specify a list of filters to collect data. By default it takes a full cluster snapshot. Data is stored in crust-gather dir by default.
+Crust gather accepts a pre-defined filter configs for Rancher or child cluster. Kubeconfig should point to the correct cluster.
+Usage with https://gist.github.com/Danil-Grigorev/0a1e629090772330bf817176ed99f324[config] file:
+
+[source,bash]
+----
+kubectl crust-gather collect-from-config -c config.yaml
+----
+
+Usage via regular flags:
+
+[source,bash]
+----
+kubectl crust-gather collect --include-namespace rancher-turtles-system --include-namespace capi-* --include-namespace cattle* --include-namespace c-* --include-namespace=<any-capi-cluster-namespace> --kubeconfig=<KUBECONFIG>
+----
+
+You can specify a file with secrets or environment variables with secrets strings to exclude. 
+
+For example:
+
+[source,bash]
+----
+kubectl crust-gather collect -s ENV_WITH_SECRET --secrets-file=secrets.txt
+----
+
+Or exclude all secret resources from collection: 
+
+[source,bash]
+----
+kubectl crust-gather collect --exclude-kind Secret
+----
+
+== How to clean up Rancher Turtles and CAPI resources
+
+Sometimes cleanup of your infrastructure might fail and it might lead to pending resources. In this situation, we have to remove resources manually.
```

Comment:
```suggestion
Sometimes cleanup of your infrastructure might fail and it might lead to pending resources. In this situation, we have to remove resources manually.

Keep in mind, that removal of finalizers manually **will** require to perform manual cleanup of provisioned resources by the infrastructure provider.
```

Reply:
awesome